frontend/src/concepts/modelRegistry/utils.ts

```diff
@@ -0,0 +1,39 @@
+export type ObjectStorageFields = {
+  endpoint: string;
+  bucket: string;
+  region?: string;
+  path: string;
+};
+
+export const objectStorageFieldsToUri = (fields: ObjectStorageFields): string | null => {
+  const { endpoint, bucket, region, path } = fields;
+  if (!endpoint || !bucket || !path) {
+    return null;
+  }
+  const searchParams = new URLSearchParams();
+  searchParams.set('endpoint', endpoint);
+  if (region) {
+    searchParams.set('defaultRegion', region);
+  }
+  return `s3://${bucket}/${path}?${searchParams.toString()}`;
+};
+
+export const uriToObjectStorageFields = (uri: string): ObjectStorageFields | null => {
```

Comment:
Can this method take `undefined` uri as a prop and return `null`? Sometimes we might want to use it like `uriToObjectStorageFields(modelArtifact.uri)` while this could be `undefined`.

Reply:
It's also OK to keep it as it it now, I can do the null check before using this function.

Reply:
Is `modelArtifact.uri` marked as optional? if so you can check emptyness BEFORE executing the function. IMO we should make the function agnostic to the attribute we are gonna pass, we are converting a uri (string) if the parameter can  be optional we do the checks before the function

Reply:
Ah, sorry I missed this before merge. Feel free to make that change in your PR when you call it if you want, or I can include that in my next PR